### ISSUE TITLE
download_data.py fails if dryrun log filename contains uppercase characters

### ISSUE THREAD

**Issue by ihough** [geoschem/geos-chem#2633]
### Your name

Ian Hough

### Your affiliation

Université Grenoble Alpes

### What happened? What did you expect to happen?

When using the `download_data.py` script to download input data from a dryrun simulation, the script fails if the dryrun log filename contains uppercase characters.

### What are the steps to reproduce the bug?

To reproduce, from a configured run dir on a case-sensitive file system:

```bash
./gcclassic --dryrun | tee Log.dryrun
./download_data.py Log.dryrun washu
```

The output will be something like:

```
Traceback (most recent call last):
  File "/home/PROJECTS/pr-geoschem/gcclassic-runs/gc_4x5_merra2_fullchem/./download_data.py", line 782, in <module>
    main()
    ~~~~^^
  File "/home/PROJECTS/pr-geoschem/gcclassic-runs/gc_4x5_merra2_fullchem/./download_data.py", line 778, in main
    download_the_data(parse_args())
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^
  File "/home/PROJECTS/pr-geoschem/gcclassic-runs/gc_4x5_merra2_fullchem/./download_data.py", line 661, in download_the_data
    paths = extract_pathnames_from_log(args)
  File "/home/PROJECTS/pr-geoschem/gcclassic-runs/gc_4x5_merra2_fullchem/./download_data.py", line 110, in extract_pathnames_from_log
    with open(dryrun_log, "r", encoding="UTF-8") as ifile:
         ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'log.dryrun'
```

### What GCClassic version were you using?

14.4.3

### What environment were you running GCClassic on?

Local cluster

### What compiler and version were you using?

gcc 14.1.0, gfortran 14.1.0

### Will you be addressing this bug yourself?

Yes

**Comment by yantosca** [comment]
Note: This is fixed by #2508, which will ship with GEOS-Chem 14.5.1.  PR #2508 will be closed once the dev/no-diff-to-benchmark is merged into main (I forgot to change the branch to dev/no-diff-to-benchmark).  We can close this as completed.
